src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -675,8 +675,12 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
                 new_schema,
                 schema_to_cast,
                 read_file_transformations,
-                _,  # we don't check for error in case of infer schema failures. We use $1, Variant type
+                e,  # we don't check for error in case of infer schema failures. We use $1, Variant type except for FileNotFoundError
```

Comment:
why change this?

Reply:
From PR description above: We need to raise Exception for FileNotFoundError in _read_semi_structured_file, needed for test/integ/modin/io/test_read_json::test_read_json_empty_file. Doesn't break anything in Snowpark Python tests/code

Reply:
Can we xfail the test and have this change in a followup PR? This one changes Snowpark Python code and better to avoid in this PR.

Reply:
I thought other changes are adding new features not modifying existing ones. This may change the existing APIs which using dateframe reader right?

Reply:
dataframe_reader.py and stored_procedure.py have changes that aren't just adding new features to make tests pass.

Reply:
I doubt they will allow us to do those in this PR but you can try.

Reply:
Discussed offline: We will separate these two changes out into other PRs and merge them into Snowpark Python